### ISSUE TITLE
Segmentation mask color flickering in model.track()

### ISSUE THREAD

**Issue by HaedamOh** [ultralytics/ultralytics#22649]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am currently testing, SAM and FastSAM 

model = SAM("sam2_t.pt") 
results = model.track(input, show=True, tracker="bytetrack.yaml") 

but I am seeing the mask visualization is flickering,
by checking at the results, mask attributes are not updated while boxes.id is updated
is results not sync with mask and boxes after running tracking part?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HaedamOh, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will follow up soon to help further.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Since you’re observing segmentation mask color flickering during `model.track()` with SAM/FastSAM, please include:
- A short code snippet showing exactly how you call `model = SAM("sam2_t.pt")` (or FastSAM) and `model.track(...)`
- One or two small input samples (image/video) that reproduce the issue
- The tracker config used (e.g. `bytetrack.yaml`) and any custom changes
- The exact `ultralytics` version, Python version, OS, and CUDA/PyTorch versions
- A screenshot or brief log of the `results` object where `boxes.id` updates but mask attributes do not
- Whether the issue occurs with both `SAM` and `FastSAM`, and the specific weights you used

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You need to use `persist=True` to retain track IDs across each inference

**Comment by HaedamOh** [comment]
Thanks, 
Do you know how can I use sam2 tracking instead of default trackers like bytetrack, bottrack

```
detector = YOLO("yolo11n.pt")
overrides = dict(conf=0.01, task="segment", mode="predict", imgsz=1024, model="sam2.1_l.pt", save=False)
predictor = SAM2DynamicInteractivePredictor(overrides=overrides, max_obj_num=30)
results = predictor(source=frame,  update_memory=True)
```
This gives me very slow around 10 seconds inference with around 5 objects in the scene.
This is slower than original SAM2 repo. 
Is this because sam2 is implemented in the same way as done in original SAM2 repo using torch compile? 
Is there any trick I can use to increase the speed?

**Comment by Y-T-G** [comment]
@HaedamOh Is it running on GPU? It shouldn't take that long.
